src/thunderbots/software/test/geom/polygon.cpp

```diff
@@ -2,29 +2,70 @@
 
 #include <gtest/gtest.h>
 
+#include <unordered_set>
+
 #include "geom/point.h"
 
-TEST(PolygonTest, test_construct)
+TEST(PolygonTest, test_construct_from_vector)
+{
+    Point p1{0.0f, 0.0f}, p2{1.0f, 0.0f}, p3{1.0f, 1.0f};
+    std::vector<Point> points{p1, p2, p3};
+    Polygon poly(points);
+
+    // check that all of the points are in the polygon
+    std::unordered_set<Point> points_set{p1, p2, p3};
+    for (const auto& p : poly.getPoints())
+    {
+        EXPECT_TRUE(points_set.find(p) != points_set.end());
+    }
+
+    // check that the segments are in the polygon
+    std::unordered_set<Segment> segments_set = {Segment{p1, p2}, Segment{p2, p3},
+                                                Segment{p3, p1}};
+    for (const auto& seg : poly.getSegments())
+    {
+        EXPECT_TRUE(segments_set.find(seg) != segments_set.end());
+    }
+}
+
+TEST(PolygonTest, test_construct_from_initializer_list)
 {
     Point p1{0.0f, 0.0f}, p2{1.0f, 0.0f}, p3{1.0f, 1.0f};
-    try
+    Polygon poly{p1, p2, p3};
+
+    // check that all of the points are in the polygon
+    std::unordered_set<Point> points_set{p1, p2, p3};
+    for (const auto& p : poly.getPoints())
     {
-        Polygon poly{Segment{p1, p2}, Segment{p2, p3}, Segment{p3, p1}};
+        EXPECT_TRUE(points_set.find(p) != points_set.end());
     }
-    catch (const std::exception& ex)
+
+    // check that the segments are in the polygon
+    std::unordered_set<Segment> segments_set = {Segment{p1, p2}, Segment{p2, p3},
+                                                Segment{p3, p1}};
+    for (const auto& seg : poly.getSegments())
     {
-        FAIL() << "Polygon constructor should not throw!";
+        EXPECT_TRUE(segments_set.find(seg) != segments_set.end());
     }
 }
 
+
 TEST(PolygonTest, test_triangle_contains_point)
```

Comment:
Probably a good idea to use `polygon` instead of `triangle` in the test name to be explicit we are testing the Polygon class and not a Triangle class.

Reply:
renamed